echidna/core/dsextract.py

```diff
@@ -0,0 +1,611 @@
+            # MC has associated trigger or vice versa and no fid v applied
+            return True
+        return False  # No associated trigger with mc or vice versa
+
+    def get_value_ntuple(self, entry):
+        '''Get radius value from an ntuple MC
+
+        Args:
+          ev (:class:`ROOT.TChain`) chain entry
+
+        Returns:
+          float: True radius
+        '''
+        return numpy.fabs(numpy.sqrt((entry.mcPosx)**2 +
+                                     (entry.mcPosy)**2 +
+                                     (entry.mcPosz)**2))
+
+
+class RadialExtractReco(Extractor):
+    '''Reconstructed radial extraction methods.
+
+    Args:
+      fv_radius (float, optional): Fiducial radius. Applies a cut to remove
+        events which have a radial position greater than the radius of the
+        fiducial volume. If None no cut is applied.
+    '''
+
+    def __init__(self, fv_radius=None):
+        '''Initialise the class
+        '''
+        super(RadialExtractReco, self).__init__("radial_reco", fv_radius)
+
+    def get_valid_root(self, ev):
+        '''Check whether radius of a DS::EV is valid
+
+        Args:
+          ev (:class:`RAT.DS.EV`) event
+
+        Returns:
+          bool: Validity boolean
+        '''
+        if ev.DefaultFitVertexExists() and \
+                ev.GetDefaultFitVertex().ContainsPosition() \
+                and ev.GetDefaultFitVertex().ValidPosition():
+            return True
+        return False
+
+    def get_value_root(self, ev):
+        '''Get radius value from a DS::EV
+
+        Args:
+          ev (:class:`RAT.DS.EV`) event
+
+        Returns:
+          float: Reconstructed radius
+        '''
+        return ev.GetDefaultFitVertex().GetPosition().Mag()
+
+    def get_valid_ntuple(self, entry):
+        '''Check whether radius of an ntuple EV is valid
+
+        Args:
+          ev (:class:`ROOT.TChain`) chain entry
+
+        Returns:
+          bool: Validity boolean
+        '''
+        if entry.scintFit == 1:
+            if self._fv_radius:
+                if numpy.fabs(numpy.sqrt((entry.posx)**2 +
+                                         (entry.posy)**2 +
+                                         (entry.posz)**2)) < self._fv_radius:
+                    return True  # Passes fiducial volume cut
+                return False  # Fails fiducial volume cut
+            return True  # Passes scintFit and no fid v cut
+        return False  # Fails scintFit
+
+    def get_value_ntuple(self, entry):
+        '''Get radius value from an ntuple EV
+
+        Args:
+          ev (:class:`ROOT.TChain`) chain entry
+
+        Returns:
+          float: Reconstructed radius
+        '''
+        return numpy.fabs(numpy.sqrt((entry.posx)**2 +
+                                     (entry.posy)**2 +
+                                     (entry.posz)**2))
+
+
+class Radial3ExtractMC(Extractor):
+    ''' True :math:`(radius/av_radius)^3` radial extraction methods.
```

Comment:
I would prefer to use `outer_radius` as the variable name here instead of `av_radius`. The radius of the AV is a constant value 5997 mm. The purpose of this part was to bin with respect to some outer radius, which *is* the AV radius by default, but can be changed so you can consider other experiments e.g. KamLAND-Zen (inner balloon of radius 3080 mm). I know it's a subtle point but I think it would make it much easier to understand what is going on.